File: src/main/java/tool/compet/gesturedetector/MyEventInfo.java
package tool.compet.gesturedetector;

import tool.compet.core.DkFloatArrayList;

class MyEventInfo {
	boolean actionPointerUp;
	boolean actionPointerDown;
	boolean actionCancel;
	boolean actionUp;
	boolean actionMove;
	boolean actionDown;
	// Stream status
	boolean configChanged;
	boolean streamCompleted;

	// Pointer
	int pointerCount; // Raw pointer count
	int skipPointerIndex;
	float divPointerCount; // For divide (pointer count after skip up-pointer)

	// Coordinate of pointers
	final DkFloatArrayList pxList = new DkFloatArrayList();
	final DkFloatArrayList pyList = new DkFloatArrayList();

	float pivotX;
	float pivotY;
}

File: tool/compet/gesturedetector/MyEventInfo.kt
package tool.compet.gesturedetector

import tool.compet.core.DkFloatArrayList

class MyEventInfo {
	var actionPointerUp = false
	var actionPointerDown = false
	var actionCancel = false
	var actionUp = false
	var actionMove = false
	var actionDown = false

	// Stream status
	var configChanged = false
	var streamCompleted = false

	// Pointer
	// Raw pointer count
	var pointerCount = 0
	var skipPointerIndex = 0

	// For divide (pointer count after skip up-pointer)
	var divPointerCount = 0f

	// Coordinate of pointers
	val pxList = DkFloatArrayList()
	val pyList = DkFloatArrayList()
	var pivotX = 0f
	var pivotY = 0f
}